### PR TITLE
MolDataset -> MolMapDataset and MolIterDataset

### PR DESCRIPTION
**Pull request by drewnutt**
Created a MolIterDataset which uses the ExampleProvider to provide a dataset with all the ExampleProviderSettings.

MolIterDataset is designed to be used with `torch.utils.data.Dataloader(batch_size=None)` since the ExampleProvider already provides a batch. Similarly to MolMapDataset, it returns a tuple of: lengths, centers, coords, atom_types, radii, and labels. These can be passed directly to `gridmaker.forward` to voxelize the batch. Translation and rotation of the examples is handled by the Dataset.

### CODE REVIEW

**Review by dkoes — APPROVED** [review]
Happy to see a test was provided.
